org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/loglines/ScanningLogFinder.xtend

```diff
@@ -24,69 +27,101 @@ import static extension java.nio.file.Files.readAllLines
  */
 class ScanningLogFinder implements LogFinder {
 
-	static val ROOT_ID = 'IDROOT'
-	static val MARKER_REGEX = Pattern.compile('''@[A-Z_]+:(ENTER|LEAVE):[0-9a-f]+:.+''')
-	static val ENTER_REGEX = Pattern.compile('''@[A-Z_]+:ENTER:[0-9a-f]+:(.+)''')
+	static val MARKER_REGEX = Pattern.compile('''@[A-Z_]+:(ENTER|LEAVE):(([0-9a-f]+:.+)|(\w+\.\w+\.\w+))''')
+	static val ENTER_REGEX = Pattern.compile('''@[A-Z_]+:ENTER:([0-9a-f]+:)?(.+)''')
 	static val ILLEGAL_TEST_EXECUTION_KEY_MESSAGE = "Provided test execution key must contain a test suite id and a test suite run id. (Key was: '%s'.)"
 
 	static val logger = LoggerFactory.getLogger(ScanningLogFinder)
 
-	@Inject extension WorkspaceProvider
+	@Inject WorkspaceProvider workspaceProvider
 	@Inject extension PersistenceConfiguration
 	@Inject extension HierarchicalLineSkipper
 	@Inject extension LogFilter
-	
-	
-	override getLogLinesForTestStep(TestExecutionKey key) {
-		return getLogLinesForTestStep(key, LogLevel.TRACE)
-	}
 
-	override getLogLinesForTestStep(TestExecutionKey key, LogLevel logLevel) {
-		Validate.notBlank(key?.suiteId, ILLEGAL_TEST_EXECUTION_KEY_MESSAGE, key?.toString)
-		Validate.notBlank(key?.suiteRunId, ILLEGAL_TEST_EXECUTION_KEY_MESSAGE, key?.toString)
+	@FinalFieldsConstructor
+	static class LogLineSelector {
+
+		protected val TestExecutionKey key
+		protected val extension WorkspaceProvider
+
+		def String[] getRelevantLines() {
+			return key.logFile.lines(UTF_8).findStart.findEnd.collect(Collectors.toList)
+		}
+
+		protected def Stream<String> findStart(Stream<String> lines) { return lines }
```

Comment:
/** default behaviour does not limit the start */

Reply:
(but I'm going with AE _behavior_ :smile: )